### ISSUE TITLE
Value of checkbox not saved to boolean property in ModelState if value didn't change

### ISSUE THREAD

**Issue by wobuzhudao** [aspnet/Mvc#2487]
When inspecting the ModelState after a POST, a Boolean property will only have the value/raw value corresponding to the state of a checkbox - only if it the state actually changed. Otherwise, it will have "true,false" for the value, and a `string[] {"true", "false"}` for the RawValue.

![untitled](https://cloud.githubusercontent.com/assets/1863756/7438794/44f4029c-f01f-11e4-9eb6-1c866bc238b1.png)

**Comment by dougbu** [comment]
@wobuzhudao please provide more information about your scenario e.g. the content of your action, view, and the model type. I suspect what you're seeing is by design since the raw value `string[] {"true", "false"}` should bind a `bool` property to `true`.  the form data identifies the state of the fields, not what has changed, and I'm assuming the property was originally `true` in this case.

**Comment by wobuzhudao** [comment]
@dougbu Ok. I tested it out some more, and you're right it is using `string[] {"true", "false"}` as the raw value whenever the checkbox is checked (whether I changed it or not).

However, whenever the checkbox is unchecked, the raw value is `"false"` (just a string, and is the same as the attempted value). Is this also by design?

I will work on a simplified version of the scenario to post here later.
